cpp/src/Ice/Incoming.cpp

```diff
@@ -1,672 +0,0 @@
-      _format(Ice::FormatType::DefaultFormat),
-      _os(instance, Ice::currentProtocolEncoding),
-      _responseHandler(std::move(responseHandler)),
-      _is(nullptr)
-{
-    _current.adapter = std::move(adapter);
-    _current.con = std::move(connection);
-    _current.requestId = requestId;
-    _current.encoding.major = 0;
-    _current.encoding.minor = 0;
-}
-
-Incoming::Incoming(Incoming&& other)
-    : _current(std::move(other._current)),
-      _servant(std::move(other._servant)),
-      _locator(std::move(other._locator)),
-      _cookie(std::move(other._cookie)),
-      _isTwoWay(std::move(other._isTwoWay)),
-      _compress(std::move(other._compress)),
-      _format(std::move(other._format)),
-      _os(other._os.instance(), Ice::currentProtocolEncoding), // TODO: make movable
-      _responseHandler(std::move(other._responseHandler)),
-      _is(std::move(other._is)) // just copies the pointer
-{
-    _observer.adopt(other._observer); // TODO: make movable
-}
-
-OutputStream*
-Incoming::startWriteParams()
-{
-    if (!_isTwoWay)
-    {
-        throw MarshalException(__FILE__, __LINE__, "can't marshal out parameters for oneway dispatch");
-    }
-
-    assert(_current.encoding >= Ice::Encoding_1_0); // Encoding for reply is known.
-
-    _os.writeBlob(replyHdr, sizeof(replyHdr));
-    _os.write(_current.requestId);
-    _os.write(replyOK);
-    _os.startEncapsulation(_current.encoding, _format);
-    return &_os;
-}
-
-void
-Incoming::endWriteParams()
-{
-    if (_isTwoWay)
-    {
-        _os.endEncapsulation();
-    }
-}
-
-void
-Incoming::writeEmptyParams()
-{
-    if (_isTwoWay)
-    {
-        assert(_current.encoding >= Ice::Encoding_1_0); // Encoding for reply is known.
-        _os.writeBlob(replyHdr, sizeof(replyHdr));
-        _os.write(_current.requestId);
-        _os.write(replyOK);
-        _os.writeEmptyEncapsulation(_current.encoding);
-    }
-}
-
-void
-Incoming::writeParamEncaps(const uint8_t* v, int32_t sz, bool ok)
-{
-    if (!ok)
-    {
-        _observer.userException();
-    }
-
-    if (_isTwoWay)
-    {
-        assert(_current.encoding >= Ice::Encoding_1_0); // Encoding for reply is known.
-        _os.writeBlob(replyHdr, sizeof(replyHdr));
-        _os.write(_current.requestId);
-        _os.write(ok ? replyOK : replyUserException);
-        if (sz == 0)
-        {
-            _os.writeEmptyEncapsulation(_current.encoding);
-        }
-        else
-        {
-            _os.writeEncapsulation(v, sz);
-        }
-    }
-}
-
-void
-Incoming::setMarshaledResult(Ice::MarshaledResult&& result)
-{
-    result.swap(_os);
-}
-
-void
-Incoming::sendResponse()
-{
-    try
-    {
-        if (_locator && !servantLocatorFinished())
-        {
-            return;
-        }
-
-        assert(_responseHandler);
-        if (_isTwoWay)
-        {
-            _observer.reply(static_cast<int32_t>(_os.b.size() - headerSize - 4));
-            _responseHandler->sendResponse(_current.requestId, &_os, _compress);
```

Comment:
The sendResponse/sendNoResponse logic is now in the new sendResponse(OutgoingResponse) function implemented by ConnectionI and CollocatedRequestHandler.